engine-nextgen/src/main/kotlin/graphql/nadel/NextgenEngine.kt

```diff
@@ -273,6 +301,24 @@ class NextgenEngine @JvmOverloads constructor(
         )
     }
 
+    private fun usesDynamicServiceResolution(
+        topLevelField: ExecutableNormalizedField
+    ): Boolean =
+        topLevelField.getFieldDefinitions(overallSchema)
+            .asSequence()
+            .filter {
+                it.getDirective(NadelDirectives.DYNAMIC_SERVICE_DIRECTIVE_DEFINITION.name) != null
+            }
+            .onEach {
+                Assert.assertTrue(GraphQLTypeUtil.unwrapNonNull(it.type) is GraphQLInterfaceType) {
```

Comment:
We have an extension function for unwrapping so you can do `it.type.unwrapNonNull()`